examples/ir/run_ir.py

```diff
@@ -0,0 +1,149 @@
+            colBERTConfig = ColBERTConfig(**args_dict)
+
+            with Run().context(RunConfig(root=args.root, experiment=args.experiment, nranks=args.nranks, amp=args.amp)):
+                trainer = Trainer(args.triples, args.queries, args.collection, colBERTConfig)
+                trainer.train(args.checkpoint)
+                model_fn = trainer.best_checkpoint_path()
+                print('model_fn: ' + model_fn)
+
+        if hasattr(process_args, 'do_index') and process_args.do_index:
+            from primeqa.ir.dense.colbert_top.colbert.indexer import Indexer
+
+            colbert_parser = Arguments(description='ColBERT indexing')
+
+            colbert_parser.add_model_parameters()
+            colbert_parser.add_model_inference_parameters()
+            colbert_parser.add_indexing_input()
+            colbert_parser.add_compressed_index_input()
+            colbert_parser.add_argument('--kmeans_niters', dest='kmeans_niters', default=20, type=int)
+            colbert_parser.add_argument('--nway', dest='nway', default=2, type=int)
+            args = colbert_parser.parse()
+
+            args_dict = vars(args)
+            # remove keys not in ColBERTConfig
+            args_dict = {key: args_dict[key] for key in args_dict if key not in ['run', 'nthreads', 'distributed', 'compression_level', 'input_arguments']}
+            # args_dict to ColBERTConfig
+            colBERTConfig = ColBERTConfig(**args_dict)
+
+            with Run().context(RunConfig(root=args.root, experiment=args.experiment, nranks=args.nranks, amp=args.amp)):
+                indexer = Indexer(args.checkpoint, colBERTConfig)
+                indexer.index(name=args.index_name, collection=args.collection, overwrite=True)
+
+        if hasattr(process_args, 'do_search') and process_args.do_search:
+            from primeqa.ir.dense.colbert_top.colbert.searcher import Searcher
+            parser = Arguments(description='ColBERT search')
+
+            parser.add_model_parameters()
+            parser.add_model_inference_parameters()
+            parser.add_compressed_index_input()
+            parser.add_ranking_input()
+            parser.add_retrieval_input()
+            args = parser.parse()
+
+            args_dict = vars(args)
+            # remove keys not in ColBERTConfig
+            args_dict = {key: args_dict[key] for key in args_dict if key not in ['run', 'nthreads', 'distributed', 'compression_level', 'qrels', 'partitions', 'retrieve_only', 'input_arguments']}
+            colBERTConfig = ColBERTConfig(**args_dict)
+
+            with Run().context(RunConfig(root=args.root, experiment=args.experiment, nranks=args.nranks, amp=args.amp)):
+                searcher = Searcher(args.index_name, checkpoint=args.checkpoint, collection=args.collection, config=colBERTConfig)
+
+                rankings = searcher.search_all(args.queries, args.topK)
+                out_fn = args.ranks_fn
+                rankings.save(out_fn)
+
```

Comment:
Depending on how large the file gets ,  may need to move each engine processing to a separate file